api/src/main/java/org/motechproject/nms/api/web/UserController.java

```diff
@@ -256,13 +257,12 @@ private UserResponse getFrontLineWorkerResponseUser(String serviceName, Long cal
             }
         }
 
-        if (flw == null) {
-            // New requirement - https://applab.atlassian.net/projects/NMS/issues/NMS-325 - Block anonymous FLWs
-            // if flw is null here, we don't already have a record from MCTS. return 403
-            if (MOBILE_ACADEMY.equals(serviceName)) {
-                throw new NotAuthorizedException(String.format(NOT_AUTHORIZED, CALLING_NUMBER));
-            }
-        } else {
+        if (MOBILE_ACADEMY.equals(serviceName)) {
+            // make sure that flw is authorized to use MA
+            restrictAnonymousMAUserCheck(flw);
```

Comment:
I was referring to the fact that the naming of the restrictAnonymousUserCheck method is bad grammar, it should be named either restrictAnonymousUser or checkRestrictedAnonymousUser.
But that's definitely http://bit.ly/1Zt2PHh